nemo/collections/nlp/data/dialogue_state_tracking_sgd/prediction_utils.py

```diff
@@ -0,0 +1,262 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+# Copyright 2019 The Google Research Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Prediction and evaluation-related utility functions.
+This file contains code artifacts adapted from the original implementation:
+https://github.com/google-research/google-research/blob/master/schema_guided_dst/baseline/pred_utils.py
+"""
+
+import json
+import os
+from collections import OrderedDict, defaultdict
+from typing import Dict, List, Optional
+
+from nemo.collections.nlp.data.dialogue_state_tracking_sgd.input_example import (
+    STATUS_ACTIVE,
+    STATUS_DONTCARE,
+    STR_DONTCARE,
+)
+from nemo.utils import logging
+
+REQ_SLOT_THRESHOLD = 0.5
+
+
+# MIN_SLOT_RELATION specifes the minimum number of relations between two slots in the training dialogues to get considered for carry-over
+MIN_SLOT_RELATION = 25
```

Comment:
How about adding REQ_SLOT_THRESHOLD  and MIN_SLOT_RELATION  to the config files to control them?

Reply:
REQ_SLOT_THRESHOLD should be 0.5 since its sigmoid threshold, I'm not using MIN_SLOT_RELATION so deleted